smac/utils/io/cmd_reader.py

```diff
@@ -10,8 +10,72 @@
 __version__ = "0.0.1"
 
 
-class CMDReader(object):
+help_type = "standard"
+
+
+class ConfigurableHelpFormatter(ArgumentDefaultsHelpFormatter):
+    """
+    Configurable Help Formatter. Can filter out developer options.
+    """
+
+    def __init(self, *args, **kwargs):
+        super(ConfigurableHelpFormatter, self).__init__(*args, **kwargs)
+
+    def _add_item(self, func, args):
+        def filter_actions(actions):
```

Comment:
could be shortened using list-comprehensions:

```
filtered_actions = [a for a in actions if isinstance(a, Action)]
# Filter out all help strings starting with '[dev]'
filtered_actions = [a for a in filtered_actions if not (isinstance(a.help, str) and a.help.startswith('[dev]'))]
# Filter out all option strings starting with '--dev'
filtered_actions = [a for a in filtered_actions if len([s for s in a.option_strings if s.startswith('--dev')])==0]
```

but i agree that there is no need to do so except for space-saving reasons. for me it's more clear using comprehensions, but i guess that's personal choice...